pennylane/capture/base_interpreter.py

```diff
@@ -587,22 +585,21 @@ def handle_while_loop(
 
 # pylint: disable=too-many-arguments
 @PlxprInterpreter.register_primitive(qnode_prim)
-def handle_qnode(self, *invals, shots, qnode, device, execution_config, qfunc_jaxpr, n_consts):
+def handle_qnode(self, *invals, shots, qnode, device, execution_config, qfunc_jaxpr):
     """Handle a qnode primitive."""
-    consts = invals[:n_consts]
-    args = invals[n_consts:]
-
-    new_qfunc_jaxpr = jaxpr_to_jaxpr(copy(self), qfunc_jaxpr, consts, *args)
 
+    qfunc_jaxpr = jaxpr_to_jaxpr(copy(self), qfunc_jaxpr, [], *invals)
+    consts = qfunc_jaxpr.consts
+    j = qfunc_jaxpr.jaxpr
+    new_qfunc_jaxpr = j.replace(constvars=(), invars=j.constvars + j.invars)
```

Comment:
Since we're basically doing the same thing over and over again, is it worth just updating `jaxpr_to_jaxpr` to do it instead?

Reply:
Actually, since we're doing it in other files as well, maybe we can add a utility function like `promote_consts(jaxpr: ClosedJaxpr) -> Jaxpr`